### PR TITLE
push: set timeout for http client

### PR DESCRIPTION
**Pull request by overvenus**
Cc #135

### CODE REVIEW

**Comment by siddontang** [comment]
LGTM

PTAL @BusyJay

**Comment by overvenus** [comment]
No @BusyJay

**Comment by BusyJay** [comment]
So can connect block forever?

**Comment by overvenus** [comment]
PTAL @BusyJay

File: src/push.rs [hunk 1]
```diff
@@ -27,15 +27,22 @@ use registry::Registry;
 use std::collections::HashMap;
 use std::hash::BuildHasher;
 use std::str::{self, FromStr};
+use std::time::Duration;
 
 const HYPER_MAX_IDLE: usize = 1;
+const HYPER_TIMEOUT_SEC: u64 = 3;
```
Comment (BusyJay) [inline src/push.rs]: Is it large enough? Can you make it configurable?

**Comment by overvenus** [comment]
PTAL

**Comment by siddontang** [comment]
PTAL @BusyJay

**Comment by BusyJay** [comment]
LGTM

**Pull request merged**